client/app/bundles/course/assessment/submission/components/VoiceResponseAnswer.jsx

```diff
@@ -0,0 +1,228 @@
+          <div>
+            <FormattedMessage {...translations.chooseVoiceFileExplain} />
+          </div>
+          {fileName}
+        </div>
+      </div>
+    );
+  }
+
+  renderAudio = (field) => {
+    const { input: { value } } = field;
+    const { file, url } = value;
+    let finalUrl;
+    if (file) {
+      finalUrl = URL.createObjectURL(file);
+    } else if (url) {
+      finalUrl = url;
+    }
+    if (finalUrl) {
+      return (<audio
+        controls
+        src={finalUrl}
+      >
+        <track kind="captions" />
+      </audio>);
+    }
+    return null;
+  }
+
+  renderAudioInput = (readOnly, recording, recordingComponentId, field) => {
+    if (readOnly) {
+      return null;
+    }
+    const { intl } = this.props;
+    return (
+      <div>
+        <div style={styles.fileInputWrapper}>
+          <SingleFileInput
+            disabled={readOnly}
+            isDropFileSupported={this.isDropFileSupported}
+            {...field}
+          >
+            {this.renderSingleFileInputChildren(field)}
+          </SingleFileInput>
+        </div>
+        <div>
+          <FlatButton
+            primary
+            className="btn btn-primary"
+            disabled={recording}
+            icon={<MicIcon />}
+            label={intl.formatMessage(translations.startRecord)}
+            onTouchTap={this.onStartRecord}
+          />
+          <FlatButton
+            primary
+            className="btn"
+            secondary
+            icon={<StopIcon />}
+            label={intl.formatMessage(translations.stopRecord)}
+            disabled={!recording || recordingComponentId !== this.currentRecordingComponentId()}
+            onTouchTap={this.onStopRecord(field)}
+          />
+        </div>
+      </div>
+    );
+  }
+
+  renderFile = ({ readOnly, recording, recordingComponentId, ...field }) => {
+    const error = field && field.meta && field.meta.error;
+    const touched = field && field.meta && field.meta.touched;
+    return (
+      <div>
+        {this.renderAudioInput(readOnly, recording, recordingComponentId, field)}
+        {this.renderAudio(field)}
+        { touched && error ? <div style={styles.errorStyle}><FormattedMessage {...error} /></div> : null }
+      </div>
+    );
+  }
+
+  render() {
+    const { question, recording, recordingComponentId, readOnly, answerId } = this.props;
+    return (
+      <div>
+        <Field
+          name={`${answerId}.file`}
+          readOnly={readOnly}
+          answerId={answerId}
+          recording={recording}
+          recordingComponentId={recordingComponentId}
+          question={question}
+          component={this.renderFile}
+        />
+      </div>
+    );
+  }
+}
+
+
+function mapStateToProps(state) {
+  return {
+    recording: state.recorder.recording,
+    recordingComponentId: state.recorder.recordingComponentId,
+  };
+}
+
+export default connect(mapStateToProps)(injectIntl(VoiceResponseAnswer));
```

Comment:
This component is connected to the Redux store. It should be placed in `containers` folder.

Reply:
Ok, I will move that. But is it a convention to do this, because there are many components connected to the store are not placed in the containers folder. This component is only related to ```Answer```, and it is only used in the ```Answer.jsx```.